design/yugabytedb-rook-design.md

```diff
@@ -0,0 +1,85 @@
+
+## Sample
+
+```yaml
+apiVersion: yugabytedb.rook.io/v1alpha1
+kind: Cluster
+metadata:
+  name: rook-yugabytedb
+  namespace: rook-yugabytedb
+spec:
+  # Mentioning network ports is optional. If some or all ports are not specified, then they will be defaulted to below-mentioned values, except for tserver-ui.
+  # For tserver-ui a cluster ip service will be created if the yb-tserver-ui port is explicitly mentioned. If it is not specified, only StatefulSet & headless service will be created for TServer. TServer ClusterIP service creation will be skipped. Whereas for Master, all 3 kubernetes objects will always be created.
+  network:
+    ports:
+      - name: yb-master-ui
+        port: 7000          # default value
+      - name: yb-master-grpc
+        port: 7100          # default value
+      - name: yb-tserver-ui
+        port: 9000          # default value
+      - name: yb-tserver-grpc
+        port: 9100          # default value
+      - name: yb-tserver-cassandra
+        port: 9042          # default value
+      - name: yb-tserver-redis
+        port: 6379          # default value
+      - name: yb-tserver-postgres
+        port: 5433          # default value
+  # Replica count for master & tserver
+  replicas:
+    master: 3
+    tserver: 3
+  scope:
+    ## Volume claim templates for Master & TServer pods
+    master:
+      volumeClaimTemplates:
+      - metadata:
+          name: datadir
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          resources:
+            requests:
+              storage: 1Gi
+          storageClassName: standard
+    tserver:
+      volumeClaimTemplates:
+      - metadata:
+          name: datadir
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          resources:
+            requests:
+              storage: 1Gi
+          storageClassName: standard
+```
+# Cluster Settings
+
+### Network
+`network` field accepts `NetworkSpec` to be specified which describes YugabyteDB network settings. This is an **optional** field. Default network settings will be used, if any or all of the acceptable values are absent.
+
+A ClusterIP service will be created when `yb-tserver-ui` port is explicitly specified. If it is not specified, only StatefulSet & headless service will be created for TServer. ClusterIP service creation will be skipped. Whereas for Master, all 3 kubernetes objects will always be created.
+
+The acceptable port names & their default values are as follows:
+
+| Name | Default Value |
+| ---- | ------------- |
+| yb-master-ui | 7000 |
+| yb-master-grpc | 7100 |
+| yb-tserver-ui | 9000 |
+| yb-tserver-grpc | 9100 |
+| yb-tserver-cassandra | 9042 |
+| yb-tserver-redis | 6379 |
+| yb-tserver-postgres | 5433 |
+
+
+### Replica Count
+Specify replica count for `master` & `tserver` pods under `replicas` field. This is a **required** field.
```

Comment:
- Would different values for `replicas` ever be used? If not, perhaps a single value would make sense?
- How should the user decide the value for replicas? Is it about how much tolerance to losing a node you could have? For example, if replicas is 3 does that mean two nodes could go down? Or does it require majority? Or something else? More details would help newcomers like me.

Reply:
- Yes the master and tserver could have different replica count
- Added link to the doc for replication and setting replica number.